pkg/event/processor.go

```diff
@@ -123,6 +128,11 @@ func NewBatchEventProcessor(options ...BPOptionConfig) *BatchEventProcessor {
 		p.BatchSize = DefaultBatchSize
 	}
 
+	if p.BatchSize > p.MaxQueueSize {
+		pLogger.Warning("Batch size is larger than queue size.  Swapping")
+		p.BatchSize, p.MaxQueueSize = p.MaxQueueSize, p.BatchSize
```

Comment:
This is a bit sneaky and I'm not sure we want to swap parameters like this.

Reply:
actually, it was a lint suggestion.  i love it!  :)